static/src/js/components/TemporalDisplay/__tests__/TemporalSelectionDropdown.test.js

```diff
@@ -291,4 +309,321 @@ describe('TemporalSelectionDropdown component', () => {
+      value: '{"startDate":"2019-03-29T00:00:00.000Z","endDate":"2019-03-30T23:59:59.999Z","isRecurring":false}'
+    })
+  })
+
+  test('onMetricsTemporalFilter is called on Clear', async () => {
+    const onMetricsTemporalFilterMock = jest.fn()
+    const user = userEvent.setup()
+
+    setup({ onMetricsTemporalFilter: onMetricsTemporalFilterMock })
+
+    await waitFor(async () => {
+      await user.click(screen.getByRole('button', { name: 'Open temporal filters' }))
+    })
+
+    const clearBtn = screen.getAllByRole('button', { name: 'Clear' }).at(2)
+    await waitFor(async () => {
+      await user.click(clearBtn)
+    })
+
+    expect(onMetricsTemporalFilterMock).toHaveBeenCalledTimes(1)
+    expect(onMetricsTemporalFilterMock).toHaveBeenCalledWith({
+      type: 'Clear Temporal Filter',
+      value: {}
+    })
+  })
+
+  test('onMetricsTemporalFilter is not called on Clear if is null', async () => {
+    const onMetricsTemporalFilterSpy = jest.spyOn(metricsActions, 'metricsTemporalFilter')
+    const user = userEvent.setup()
+
+    setup({ onMetricsTemporalFilter: null })
+
+    await waitFor(async () => {
+      await user.click(screen.getByRole('button', { name: 'Open temporal filters' }))
+    })
+
+    const clearBtn = screen.getAllByRole('button', { name: 'Clear' }).at(2)
+    await waitFor(async () => {
+      await user.click(clearBtn)
+    })
+
+    expect(onMetricsTemporalFilterSpy).toHaveBeenCalledTimes(0)
+  })
+
+  test('onMetricsTemporalFilter is called when toggling recurring', async () => {
+    const onMetricsTemporalFilterMock = jest.fn()
+    const user = userEvent.setup()
+
+    setup({ onMetricsTemporalFilter: onMetricsTemporalFilterMock })
+
+    await waitFor(async () => {
+      await user.click(screen.getByRole('button', { name: 'Open temporal filters' }))
+    })
+
+    const recurringCheckbox = screen.getByLabelText('Recurring?')
+
+    await waitFor(async () => {
+      await user.click(recurringCheckbox)
+    })
+
+    expect(onMetricsTemporalFilterMock).toHaveBeenCalledTimes(1)
+    expect(onMetricsTemporalFilterMock).toHaveBeenCalledWith({
+      type: 'Set Recurring',
+      value: true
+    })
+  })
+
+  test('onMetricsTemporalFilter is not called when it is null and recurring is toggled', async () => {
+    const onMetricsTemporalFilterSpy = jest.spyOn(metricsActions, 'metricsTemporalFilter')
+    const user = userEvent.setup()
+
+    setup({ onMetricsTemporalFilter: null })
+
+    await waitFor(async () => {
+      await user.click(screen.getByRole('button', { name: 'Open temporal filters' }))
+    })
+
+    const recurringCheckbox = screen.getByLabelText('Recurring?')
+
+    await waitFor(async () => {
+      await user.click(recurringCheckbox)
+    })
+
+    expect(onMetricsTemporalFilterSpy).toHaveBeenCalledTimes(0)
+  })
+
+  describe('setStartDate', () => {
+    beforeEach(() => {
+      TemporalSelectionDropdownMenu.mockImplementation(
+        ({ setStartDate, setEndDate }) => {
```

Comment:
This exact component is defined twice, here and the beforeEach in the describe 'setEndDate'. You can either:

1. Pull the whole component out and define it once at the top of the file. Then reference the component in both beforeEach blocks
2. Remove the end date buttons from this component and remove the start date buttons from the describe 'setEndDate' component